.github/workflows/publish.yml

```diff
@@ -0,0 +1,41 @@
+name: Publish Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: The branch to release from.
+        required: true
+        default: master
+
+permissions:
+  contents: read
+
+jobs:
+  publish:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    environment: release
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.inputs.branch }}
+
+      - name: Configure Ruby
+        uses: ./.github/actions/setup
+        with:
+          ruby: 3.2
+
+      - name: Publish to RubyGems
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem build *.gemspec
+          gem push *.gem
+        env:
+          GEM_HOST_API_KEY: "${{secrets.RUBYGEMS_AUTH_TOKEN}}"
```

Comment:
Is there a need to both create the `credentials` file _and_ create an environment variable? Wouldn't we just use the env var?

Reply:
Not familiar enough with Ruby to make an authoritative statement one way or the other on that, but this is implemented per [GitHub's suggested approach](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-ruby#publishing-gems). My assumption was the environment variable was being used elsewhere in the `gem` process.